src/core/friendly_errors/sketch_verifier.js

```diff
@@ -111,16 +164,116 @@ function sketchVerifier(p5, fn) {
     return userDefinitions;
   }
 
+  /**
+   * Checks user-defined variables and functions for conflicts with p5.js
+   * constants and global functions.
+   * 
+   * This function performs two main checks:
+   * 1. Verifies if any user definition conflicts with p5.js constants.
+   * 2. Checks if any user definition conflicts with global functions from
+   * p5.js renderer classes.
+   * 
+   * If a conflict is found, it reports a friendly error message and halts
+   * further checking.
+   * 
+   * @param {Object} userDefinitions - An object containing user-defined variables and functions.
+   * @param {Array<{name: string, line: number}>} userDefinitions.variables - Array of user-defined variable names and their line numbers.
+   * @param {Array<{name: string, line: number}>} userDefinitions.functions - Array of user-defined function names and their line numbers.
+   * @returns {boolean} - Returns true if a conflict is found, false otherwise.
+   */
+  fn.checkForConstsAndFuncs = function (userDefinitions) {
+    const allDefinitions = [
+      ...userDefinitions.variables,
+      ...userDefinitions.functions
+    ];
+
+    // Helper function that generates a friendly error message that contains
+    // the type of redefinition (constant or function), the name of the
+    // redefinition, the line number in user's code, and a link to its
+    // reference on the p5.js website.
+    function generateFriendlyError(errorType, name, line) {
+      const url = `https://p5js.org/reference/#/p5/${name}`;
+      const message = `${errorType} "${name}" on line ${line} is being redeclared and conflicts with a p5.js ${errorType.toLowerCase()}. JavaScript does not declaring a ${errorType.toLowerCase()} more than once. p5.js reference: ${url}.`;
+      return message;
+    }
+
+    // Helper function that checks if a user definition has already been defined
+    // in the p5.js library, either as a constant or as a function.
+    function checkForRedefinition(name, libValue, line, type) {
+      try {
+        const userValue = eval("name");
+        if (libValue !== userValue) {
+          let message = generateFriendlyError(type, name, line);
+          console.log(message);
+          return true;
+        }
+      } catch (e) {
+        // If eval fails, the function hasn't been redefined
+        return false;
+      }
+      return false;
+    }
+
+    // Checks for constant redefinitions.
+    for (let { name, line } of allDefinitions) {
+      const libDefinition = constants[name];
+      if (libDefinition !== undefined) {
+        if (checkForRedefinition(name, libDefinition, line, "Constant")) {
+          return true;
+        }
+      }
+    }
+
+    // Get the names of all p5.js functions which are available globally
+    const classesWithGlobalFns = ['Renderer', 'Renderer2D', 'RendererGL'];
```

Comment:
I'm actually not sure we need this any more in 2.0. I don't have the full context on why we had this initially but I think now, the only globals are things that have been added directly to `p5.prototype`. @limzykenneth does that sound right?

Reply:
Yes, these classes will not have methods attached to global. The rules for attaching anything to global are (if true for both of the following):
1. It is a member of `p5.prototype`
2. Its name does not start with `_`